Build inherits $SRC from shell

CBMC version: develop branch
Operating system: OSX
Exact command line resulting in the issue: make -C src
What behaviour did you expect: build
What happened instead: confusing, unrelated error message about something (the value of SRC) being a directory.

To reproduce

```
$ git clone cbmc-url
$ SRC=/var make -C src
## Entering big-int
/Applications/Xcode.app/Contents/Developer/usr/bin/make  -C big-int
make[1]: *** /var: Is a directory.  Stop.
make: *** [big-int.dir] Error 2
```

--- a/src/big-int/makefile
+++ b/src/big-int/makefile
@@ -1,7 +1,8 @@
+SRC = bigint-func.cc bigint.cc
+
 include ../config.inc
 include ../common
 
-SRC = bigint-func.cc bigint.cc
 CLEANFILES = big-int$(LIBEXT)
 
 all: big-int$(LIBEXT)

--- a/src/common
+++ b/src/common
@@ -270,6 +270,7 @@ sources: Makefile
 # endif
 
 D_FILES1 = $(SRC:.c=$(DEPEXT))
-D_FILES = $(D_FILES1:.cpp=$(DEPEXT))
+D_FILES2 = $(D_FILES1:.cc=$(DEPEXT))
+D_FILES = $(D_FILES2:.cpp=$(DEPEXT))
 
 -include $(D_FILES)

